nixos/modules/services/web-apps/flarum.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkOption types literalExample;
+  cfg = config.services.flarum;
+in {
+  options = {
+    services.flarum = {
+      enable = mkEnableOption "Flarum discussion platform";
+
+      packages = mkOption {
+        type = with types; attrsOf package;
+        example = literalExample "pkgs.flarum";
+        description = "Packages used for the service.";
+        default = pkgs.flarum;
+      };
+
+      domain = mkOption {
+        type = types.str;
+        default = "localhost";
+        example = "forum.example.com";
+        description = "Domain to serve on.";
+      };
+
+      installPath = mkOption {
```

Comment:
What is this needed for? It looks like the software is packaged so you don't need to install (already in `/nix/store`).

Reply:
Unfortunately, the software needs to reside in a writable file system location.

Reply:
All of the software, or just the `storage/` folder and `config.php`?

Reply:
After reviewing upstream documentation and code I think you can patch this to work with the nix store. Comments in source code even mention this. See [here](https://github.com/flarum/flarum/blob/0ef4318913df417d3dde5024a085b7f2c3f7804d/public/index.php#L10) and [here](https://github.com/flarum/flarum/blob/0ef4318913df417d3dde5024a085b7f2c3f7804d/site.php#L46).